test/DaoModule.spec.ts

```diff
@@ -854,6 +854,43 @@ describe("DaoModule", async () => {
             ).to.be.revertedWith("Cannot execute transaction again");
         })
 
+        it.only("throws if too little gas is used for the transaction", async () => {
+            const { executor, mock, module, oracle } = await setupTestWithTestExecutor();
+            await executor.setModule(module.address)
+
+            const id = "some_random_id";
+            const tx = { to: mock.address, value: 42, data: "0xbaddad", operation: 0, nonce: 0 }
+            const txHash = await module.getTransactionHash(tx.to, tx.value, tx.data, tx.operation, tx.nonce)
+            const question = await module.buildQuestion(id, [txHash]);
+            const questionId = await module.getQuestionId(1337, question, executor.address, 42, 0, 0)
+
+            await mock.givenMethodReturnUint(oracle.interface.getSighash("askQuestion"), questionId)
+            await module.addProposal(id, [txHash])
+            const block = await ethers.provider.getBlock("latest")
+            await mock.givenMethodReturnBool(oracle.interface.getSighash("resultFor"), true)
+            await mock.givenMethodReturnUint(oracle.interface.getSighash("getFinalizeTS"), block.timestamp)
+            await nextBlockTime(hre, block.timestamp + 24)
+            expect(
+                (await mock.callStatic.invocationCount()).toNumber()
+            ).to.be.equals(1)
+            //console.log(await module.estimateGas.executeProposalWithIndex(id, [txHash], tx.to, tx.value, tx.data, tx.operation, tx.nonce))
+            await logGas("Execute proposal", module.executeProposalWithIndex(id, [txHash], tx.to, tx.value, tx.data, tx.operation, tx.nonce, { gasLimit: 367600 }))
+            /*
+            await expect(
+                module.executeProposalWithIndex(id, [txHash], tx.to, tx.value, tx.data, tx.operation, tx.nonce)
+            ).to.be.revertedWith("out of gas");
+            */
+            expect(
+                (await mock.callStatic.invocationCount()).toNumber()
+            ).to.be.equals(1)
```

Comment:
No additional invocations are registered (was one before). Therefore even so the transaction failed it did not revert. Maybe it would be useful to add an integration test with the Safe contracts, as they should emit events in this case (not really solving the issue, but making it more visible)